### ISSUE TITLE
Support CREATE DEFAULT INDEX ON foo;

### ISSUE THREAD

**Issue by rjnn** [MaterializeInc/materialize#2614]
I I have a view. I created it unmaterialized. For various reasons, I cannot drop it and recreate it using `CREATE MATERIALIZED VIEW` (basically, I have downstream views and i don't want to do a cascading drop).

I want to materialize it. I want this to be simple. The syntax is fairly obvious:

`ALTER VIEW foo SET MATERIALIZED = true`

This should go ahead and create the index that it would have created in the first place, had I run `CREATE MATERIALIZED VIEW foo` rather than `CREATE VIEW foo`

**Comment by ruchirK** [comment]
can't you use `CREATE INDEX` for this functionality? or is the ask here to make that be simple?

**Comment by rjnn** [comment]
The ask here is to make it simple. I don't know what index to create.

**Comment by ruchirK** [comment]
👍

**Comment by benesch** [comment]
I'm still a bit skeptical that this syntax covers all the edge caeses. What does `ALTER VIEW v SET MATERIALIZED = false` do? In the case that you ran `CREATE MATERIALIZED VIEW v`, or `ALTER VIEW v SET MATERIALIZED = true`, the answer is obvious—delete that auto-created index—but what does it do if you ran `CREATE INDEX` manually, or created multiple indexes?

**Comment by rjnn** [comment]
I'm fine with those `set=false` not working great, but `set=true` is a common UX issue - I think I've encountered this about 10 times so far where I regret not materializing things.

Asking folks to manually drop indexes is easy: they just run `SHOW INDEXES FROM <view>` and then shoot them in the head one by one. It's okay for the automatic path to not work. But asking folks to create an index is hard: it requires thinking about the index. And, this is what we purport to be able to do as our raison d'etre! And we should make it easy for folks to access that superpower.

**Comment by frankmcsherry** [comment]
Should we just make `CREATE INDEX` easy to use?

**Comment by rjnn** [comment]
I mean its easy enough to use. If i wanted an index, I can create an index. But I have a named view, and I want it to be materialized. The code already exists for determining what indexes must be created in order to materialize the view: that's what we run when people say `CREATE MATERIALIZED VIEW`. Why not just allow people to run that code?

**Comment by frankmcsherry** [comment]
I think mostly because having a great many non-orthogonal ways to do related things moves technical burden from users on to developers.

**Comment by rjnn** [comment]
After much discussion on slack, it appears there is a fix that is considerably more palatable to engineering, which would solve the problem here.

The main objection was adding tonnes of different code paths that achieve almost the same thing. Currently the code path here is `CREATE INDEX`. All we need is a small tweak; this is what Postgres already supports with `CREATE DEFAULT INDEX`! Postgres also does not require an index name, quoting (thanks for finding this @benesch):

> The name of the index to be created. No schema name can be included here; the index is always created in the same schema as its parent table. If the name is omitted, PostgreSQL chooses a suitable name based on the parent table’s name and the indexed column name(s).

So if we simply support `CREATE DEFAULT INDEX ON foo` this would take an unmaterialized view and materialize it. Tada!

**Comment by benesch** [comment]
Oh, sorry, one small point of clarification: `CREATE INDEX ON table (col)` with the name omitted is a Postgresism, but `CREATE DEFAULT INDEX` is a Nikhilism. I think the jury is still out on whether

```sql
CREATE DEFAULT INDEX ON foo
```

or

```sql
CREATE INDEX ON foo
```

or 

```sql
CREATE INDEX ON foo DEFAULT COLUMNS
```

is preferable. But I think we are very close to consensus!

The other remaining question I have is what `CREATE INDEX ON foo` twice in a row should do. Probably create two indexes?

**Comment by benesch** [comment]
```
benesch=# create table t (a int);
CREATE TABLE
benesch=# create index on t (a);
CREATE INDEX
benesch=# create index on t (a);
CREATE INDEX
benesch=# create index on t (a);
CREATE INDEX
benesch=# 
benesch=# create index on t (a);
CREATE INDEX
```

**Comment by rjnn** [comment]
If we create multiple indexes, will they share arrangements (i.e. be a catalog duplication but a dataplane no-op?

**Comment by benesch** [comment]
Great question! I have no idea. @frankmcsherry do you know?

**Comment by rjnn** [comment]
I would like this addressed in 0.3, or 0.4. @cuongdo would you like to make a call as to which one is more appropriate?

**Comment by cuongdo** [comment]
Let's aim for 0.3.
